pkg/cdi/producer/producer.go

```diff
@@ -0,0 +1,86 @@
+/*
+   Copyright © 2024 The CDI Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package producer
+
+import (
+	"path/filepath"
+
+	cdi "tags.cncf.io/container-device-interface/specs-go"
+)
+
+// A Producer defines a structure for outputting CDI specifications.
+type Producer struct {
+	format       specFormat
+	failIfExists bool
+}
+
+// New creates a new producer with the supplied options.
+func New(opts ...Option) (*Producer, error) {
+	p := &Producer{
+		format: DefaultSpecFormat,
```

Comment:
> Do we know if this would be an allowed dependency in the k8s case ?

Just based on a quick git grep, it is an indirect dependency hack hack/tools, but not k8s itself. So I'm guessing... inconclusive.

Reply:
It seems as if it used to be a top-level import but was recently removed.